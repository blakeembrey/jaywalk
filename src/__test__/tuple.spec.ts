import { validationFunction } from '../'

describe('Tuple', () => {
  it('should validate tuples', () => {
    const validate = validationFunction({
      '@type': 'Tuple',
      tuple: [
        {
          '@type': 'String'
        },
        {
          '@type': 'Boolean'
        }
      ]
    })

    const input = ['testing', true]

    return validate(input).then(function (output) {
      expect(output).toEqual(input)
    })
  })

  it('should reject non-tuples', () => {
    const validate = validationFunction({
      '@type': 'Tuple',
      tuple: [
        {
          '@type': 'String'
        },
        {
          '@type': 'Boolean'
        }
      ]
    })

    expect.assertions(1)

    return validate(true).catch(function (err) {
      expect(err.errors).toMatchSnapshot()
    })
  })

  it('should reject incorrect tuples', () => {
    const validate = validationFunction({
      '@type': 'Tuple',
      tuple: [
        {
          '@type': 'String'
        },
        {
          '@type': 'Boolean'
        }
      ]
    })

    expect.assertions(1)

    return validate([1, 2]).catch(function (err) {
      expect(err.errors).toMatchSnapshot()
    })
  })
})
