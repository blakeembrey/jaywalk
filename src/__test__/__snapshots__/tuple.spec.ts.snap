// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tuple should reject incorrect tuples 1`] = `
Array [
  Object {
    "dataPath": Array [
      0,
    ],
    "message": "not a string",
    "schemaPath": Array [
      "tuple",
      0,
      "@type",
    ],
  },
  Object {
    "dataPath": Array [
      1,
    ],
    "message": "not a boolean",
    "schemaPath": Array [
      "tuple",
      1,
      "@type",
    ],
  },
]
`;

exports[`Tuple should reject non-tuples 1`] = `
Array [
  Object {
    "dataPath": Array [],
    "message": "not a tuple",
    "schemaPath": Array [
      "@type",
    ],
  },
]
`;
